pkg/arvo/sur/invite-store.hoon

```diff
@@ -0,0 +1,34 @@
+|%
+++  serial  @uvH
+::
++$  invite
+  $:  =path
+      =ship
+      recipient=ship
+      app=@tas
+      text=cord
+  ==
+::
++$  invitatory  (map serial invite)  ::  containing or conveying an invitation
```

Comment:
Why do we map these by serial? I'd expect to map them by ship. (ie, do uniqueness based on "ship and context" of invite, not "uid" of invite. If I invite you multiple times to `~paldev/secret`, then *maybe* you should get multiple notifications (on-receive), but there certainly shouldn't be duplicate invitations stored/managed in state.

Reply:
The `invites` path is not semantically the same type of path as the one in an individual invite.

For instance, the %chat app only wants to see invites sent to /chat, so it creates an `invitatory` at the /chat path. Then within that invitatory, invites are unique, so they each have a serial identifier.

Reply:
I should just put comments in the sur file explaining this.

Reply:
Okay, that makes sense. Still feel pretty icky about using serial identifiers though. Have you thought about doing `[=ship resource=path]` as identifiers? ie, for `/chat`, doing `[~tacryt-socryp /dm-with-tacryt]` for (the one, auto-deduplicated) invite relating to that specific chat.

Reply:
I like the serial identifiers, personally. They're unique, meaningful to the client, and duplicate invites are rare / a non-issue.

Reply:
> meaningful to the client

Wait, maybe I misunderstand, but `+$  serial  @uvH` suggests something like `0v6.bmjjf.t2l6p.h5s63.oop3a.5lc54`. How would they be meaningful?

Reply:
Meaningfully different from each other insofar as individual invites are differentiated from each other via the unique identifier, and they're already context-specific due to the `(map path invitatory` pattern.

Reply:
Like the indirection between invite and invitatory, whether this is righteous seems to depend on what invites are (for), and how clients should use these components. If you've already been invited to something, should it be a state transition if you're invited to it again?

Reply:
Likely yes, because you may have left the chatroom (specific case) or deleted the data associated with the subscription and canceled the subscription the invite had given you access to previously. Thus you’re being re-invited.